src/lib/components/DatePickerPanel.js

```diff
@@ -102,6 +115,14 @@ export default class DatePickerPanel extends React.Component {
     });
   }
 
+  validateOnTimeChange() {
+    const { currentValue, confirmDisabled } = this.state;
+    const newConfirmDisabled = (currentValue.valueOf() <= Date.now());
+    if (confirmDisabled !== newConfirmDisabled) {
+      this.setState({ confirmDisabled: newConfirmDisabled });
```

Comment:
No, I think React is lazy here - if you call `setState` an update gets queued. That state update object could have some deep structured data, so I think they leave it up to the component to decide.